packages/dtslint-runner/src/post-results.ts

```diff
@@ -0,0 +1,151 @@
+        }
+      }
+
+      newComment = `Hey @${userToTag}, the results of running the DT tests are ready.\n`;
+      const diffComment = getDiffComment(nightlyErrors, branchErrors);
+      if (diffComment) {
+        newComment += `There were interesting changes:`
+        if (newComment.length + diffComment.length + checkLogsMessage.length > 65535) {
+          newComment += `Changes are too big to display here, please check the log.`
+        }
+        else {
+          newComment += diffComment;
+        }
+      }
+      else {
+        newComment += "Everything looks the same!"
+      }
+      newComment += checkLogsMessage;
+    }
+    
+    const response = await gh.issues.createComment({
+      issue_number: +issue,
+      owner: "Microsoft",
+      repo: "TypeScript",
+      body: newComment
+    });
+
+    const newCommentUrl = response.data.html_url;
+    const statusComment = await gh.issues.getComment({
+      owner: "Microsoft",
+      repo: "TypeScript",
+      comment_id: +statusCommentId,
+    });
+
+    const newBody = `${statusComment.data.body}\n\nUpdate: [The results are in!](${newCommentUrl})`;
+    await gh.issues.updateComment({
+      owner: "Microsoft",
+      repo: "TypeScript",
+      comment_id: +statusCommentId,
+      body: newBody
+    });
+  }
+  catch (e) {
+    await gh.issues.createComment({
+      issue_number: +issue,
+      owner: "Microsoft",
+      repo: "TypeScript",
+      body: `Hey @${userToTag}, something went wrong when publishing results. ${checkLogsMessage}`
+    });
+  }
+}
+
+function getDiffComment(nightly: Errors, branch: Errors): string | undefined {
+  const nightlyMap = new Map(nightly.map(error => [error.path, error]));
+  const branchMap = new Map(branch.map(error => [error.path, error]));
+
+  const nightlyOnly = [];
+  const bothChanged = [];
+
+  for (const [path, error] of nightlyMap) {
+    if (branchMap.has(path)) {
+      const branchError = branchMap.get(path)!;
+      if (branchError.message !== error.message) {
+        bothChanged.push({ path, nightly: error.message, branch: branchError.message });
+      }
+    }
+    else {
+      nightlyOnly.push(error)
+    }
+  }
+
+  const branchOnly = [];
+  for (const [path, error] of branchMap) {
+    if (nightlyMap.has(path)) {
+      continue; // Already considered above
+    }
+    else {
+      branchOnly.push(error)
+    }
+  }
+
+  if (!nightlyOnly.length && !bothChanged.length && !branchOnly.length) {
+    return undefined;
+  }
+
+  const nightlyOnlyMessage =
+    nightlyOnly.length ? `
+<details>
+<summary>Nightly only errors:</summary>
+${nightlyOnly.map(err => `Package: ${err.path}\nError: ${err.message}`).join("\n")}
+</details>
+` : "";
+  const bothChangedMessage =
+    bothChanged.length ? `
+<details>
+<summary>Errors that changed:</summary>
+${bothChanged.map(err => `Package: ${err.path}\nNightly error: ${err.nightly}\nBranch error: ${err.branch}`).join("\n")}
+</details>
+` : "";
+  const branchOnlyMessage =
```

Comment:
these are likely to be the actually valuable errors. Maybe this should be first, and not inside a `details` tag if there are only 1 - 3 errors?